app/serializers/job_application_serializer.rb

```diff
@@ -6,11 +6,23 @@ class JobApplicationSerializer
              :notes, 
              :job_description, 
              :application_url, 
-             :contact_information, 
              :company_id,
              :company_name
              
   attribute :company_name do |job_application|
     job_application.company&.name
   end
+
+  def self.contacts_for(job_application)
+    job_application.user.contacts.where(company_id: job_application.company_id).map do |contact|
```

Comment:
I really like how this is structured, very clean. The ORM language and iterating work nicely.